test/no-browser-sniffing.test.ts

```diff
@@ -0,0 +1,115 @@
+import connect, { NextFunction } from 'connect';
+import request from 'supertest';
+import { IncomingMessage, ServerResponse } from 'http';
+
+import parseCsp from 'content-security-policy-parser';
+
+import AGENTS from './browser-data.json';
+
+import csp = require('..');
+
+const POLICY = {
+  defaultSrc: ["'self'"],
+  'script-src': ['scripts.biz'],
+  styleSrc: [
+    'styles.biz', function (_req: IncomingMessage, res: ServerResponse) {
```

Comment:
Is there a way to type this as a "superset" of `IncomingMessage` so that we don't have to cast it to `any`? If not, is there another way to do this so we don't have to do this cast?

If there's a way to clean this up, we should make a related change in `makeApp` below, and similarly in `normal-browsers.test.ts`.

Reply:
Have replaced `any` with a safer cast to `ExtendedServerResponse`, a custom interface that extends `ServerResponse`.